draft-ietf-mls-protocol.md

```diff
@@ -4441,72 +4441,26 @@ current state of the group.  In practice, however, there is a risk
 that two members will generate Commit messages simultaneously,
 based on the same state.
 
-When this happens, there is a need for the members of the group to
-deconflict the simultaneous Commit messages.  There are two
-general approaches:
-
-* Have the Delivery Service enforce a total order
-* Have a signal in the message that clients can use to break ties
-
-As long as Commit messages cannot be merged, there is a risk of
-starvation.  In a sufficiently busy group, a given member may never
-be able to send a Commit message, because he always loses to other
-members.  The degree to which this is a practical problem will depend
+Applications MUST have an established way to resolve conflicting Commit messages
+for the same epoch. They can do this either by preventing conflicting messages
+from occurring in the first place, or by developing rules for deciding which
+Commit out of several sent in an epoch will be canonical. The approach chosen
+MUST minimize the amount of time that forked or previous group states are kept
+in memory, and promptly delete them once they're no longer necessary to ensure
+forward secrecy.
+
+The generation of Commit messages MUST NOT modify a client's state, since the
+client doesn't know at that time whether the changes implied by the Commit
+message will conflict with another Commit or not. Similarly, a client MUST NOT
+send Welcome messages corresponding to a Commit until it's clear that the Commit
+has been accepted.
```

Comment:
Interim 2022-06-02:

* Probably too tight.  E.g., client could send Commit+Welcome to DS, then DS could decide whether to send Welcome based on whether Commit is accepted.  Proposed text:

> The Welcome message corresponding to a Commit MUST NOT be delivered to a new joiner until it's clear that the Commit has been accepted.

* Otherwise clear to merge